libp2p/network/url_conversion_test.go

```diff
@@ -0,0 +1,159 @@
+package network
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/multiformats/go-multiaddr"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPeerMultiAddrFromServiceURL_success(t *testing.T) {
+	testCases := []struct {
+		name                  string
+		serviceUrl            string
+		expetedMultiaddrRegex string
+	}{
+		{
+			"FQDN",
+			"www.google.com:8080",
+			`/ip4/(\d{1,3}\.){3}\d{1,3}/tcp/8080`,
+		},
+		{
+			"IPv4",
+			"142.250.181.196:8080",
+			"/ip4/142.250.181.196/tcp/8080",
+		},
+		{
+			"IPv6",
+			"2a00:1450:4005:802::2004:8080",
+			"/ip6/2a00:1450:4005:802::2004/tcp/8080",
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			actualMultiaddr, err := Libp2pMultiaddrFromServiceUrl(testCase.serviceUrl)
+			require.NoError(t, err)
+			require.NotNil(t, actualMultiaddr)
+			require.Regexp(t, regexp.MustCompile(testCase.expetedMultiaddrRegex), actualMultiaddr.String())
+		})
+	}
+}
+
+const (
```

Comment:
Where are we using these?

Reply:
Just in this test. I'll unexport them. :wink: